### ISSUE TITLE
Reloading Help buffer load DuckDuckGo

### ISSUE THREAD

**Issue by Daniel-V1** [atlas-engineer/nyxt#745]
**Describe the bug**
When in the initial buffer or tutorial buffer, reloading the page instead loads DuckDuckGo. 
I learned this when I was trying to see if loading a mode and then reloading the tutorial page would give me the new keybindings. 

**Precise recipe to reproduce the issue**
Open next, run command reload-current-buffer

**Information**
- Graphics card and driver: Nvidia, Nvidia
- Desktop environment / Window manager name+version: i3 version 4.18.1 (2020-04-23) © 2009 Michael Stapelberg and contributors
- How you installed Next (Guix pack, package manager, build from source): Source
- Information from command copy-system-information: 
Next Version: 1.5.0-9ccda09c 
Lisp Implementation: SBCL
Lisp Version: 2.0.3 
Operating System: Linux 5.6.13-arch1-1 
Features: (WEBKIT2 WEBKIT2-2.28 WEBKIT2-EMOJI WEBKIT2-MEDIA WEBKIT2-SANDBOXING
           GTK-3-22 GTK-3-20 GTK-3-18 GTK-3-16 GTK-3-14 GTK-3-12 GTK-3-10
           GTK-3-8 GTK-3-6 GTK-3-4 GTK GDK-3-22 GDK-3-20 GDK-3-18 GDK-3-16
           GDK-3-14 GDK-3-12 GDK-3-10 GDK-3-8 GDK-3-6 GDK-3-4 CAIRO-1-10
           CAIRO-1-12 GDK-PIXBUF GLIB-2-30 GLIB-2-32 GLIB-2-34 GLIB-2-36
           GLIB-2-38 GLIB-2-40 GLIB-2-42 GLIB-2-44 GLIB-2-46 GLIB-2-48
           GLIB-2-50 GLIB-2-52 GLIB-2-54 GLIB-2-56 GLIB-2-58 GLIB
           FSET-EXT-STRINGS SWANK PLUMP-UTF-32 GLOBAL-VARS DECLARE-TYPES
           PARENSCRIPT NAMED-READTABLES OSICAT-FD-STREAMS LPARALLEL 21BIT-CHARS
           CL-FAD CHUNGA FLEXI-STREAMS CLOSER-MOP CL-PPCRE-UNICODE CL-UNICODE
           CL-PPCRE BORDEAUX-THREADS LPARALLEL.WITH-CLTL2 LPARALLEL.WITH-CAS
           LPARALLEL.WITH-STEALING-SCHEDULER SPLIT-SEQUENCE GRAY-STREAMS
           FLAT-NAMESPACE X86-64 UNIX CFFI FLAT-NAMESPACE SEQUENCE-EMPTYP
           FAST-IO-SV FAST-IO SBCL-USES-SB-ROTATE-BYTE ASDF-SYSTEM-CONNECTIONS
           CL-JSON-CLOS CL-JSON THREAD-SUPPORT QUICKLISP
           SB-BSD-SOCKETS-ADDRINFO ASDF3.3 ASDF3.2 ASDF3.1 ASDF3 ASDF2 ASDF
           OS-UNIX NON-BASE-CHARS-EXIST-P ASDF-UNICODE X86-64 GENCGC 64-BIT
           ANSI-CL COMMON-LISP ELF IEEE-FLOATING-POINT LINUX LITTLE-ENDIAN
           PACKAGE-LOCAL-NICKNAMES SB-CORE-COMPRESSION SB-LDB SB-PACKAGE-LOCKS
           SB-THREAD SB-UNICODE SBCL UNIX)

**Output when started from a shell**
Next version 1.5.0-9ccda09c
<INFO> [16:39:25] Using data profile "default".
<INFO> [16:39:25] Listening to socket "/home/daniel/.local/share/next/next.socket".
<INFO> [16:39:43] Loading global history of 11 URLs from "/home/daniel/.local/share/next/history.lisp".
<INFO> [16:39:43] Saving session to "/home/daniel/.local/share/next/sessions/default.lisp".
<INFO> [16:39:43] Loading "https://duckduckgo.com/?q=".
<INFO> [16:39:43] Saving session to "/home/daniel/.local/share/next/sessions/default.lisp".
<INFO> [16:39:45] Finished loading "https://duckduckgo.com/?q=".

**Comment by Ambrevar** [comment]
Thanks for reporting!

Indeed, it's high time we implemented a `next://...` protocol to load
Next-specific pages.  This way we could start hyperlinking the
docstrings between each other, or the "tutorial" page from the "help"
page for instance.

Any idea how other browsers to it?

- The special pages' URL must be set (e.g. next://tutorial).

- Some function needs to understand `next://`, maybe `parse-url'?

- The docstrings hyplerlinks must be parse, e.g. `foo' to `next://foo'.
  But maybe that's not enough because we would need a way to refer to
  class slots without ambiguity.  We also need to be able to refer to
  a variable and a function of the same name without ambiguity.
  Is there an idiomatic syntax for that in Common Lisp docstrings?
  If not, shall we come up with one?  We should not make the docstrings
  less idiomatic in the code.

**Comment by Daniel-V1** [comment]
It sounds to me like you've got the right ideas. For firefox it looks like extensions use a moz-extension: scheme, and for configuration they've got the about: scheme.
I don't know enough about the internals of Next unfortunately, but parse-url sounds like a reasonable place to do the scheme dispatch.
I searched a little bit for more information on docstring conventions, but didn't really find much for Common Lisp. I did find a reddit thread pointing me at the Emacs Lisp manual for docstring conventions though, and it looks like they just use quotes and keywords before them to disambiguate, so for class slots maybe just slot 'search-engines' or something. 
Thank you so much for your work on this project!

https://www.gnu.org/software/emacs/manual/html_node/elisp/Documentation-Tips.html#Documentation-Tips
